enhancements/cluster-version-operator/upgrades.md

```diff
@@ -0,0 +1,83 @@
+# Upgrades and order
+
+In the CVO, upgrades are performed in the order described in the payload (lexographic), with operators only running in parallel if they share the same file prefix `0000_NN_` and differ by the next chunk `OPERATOR`, i.e. `0000_70_authentication-operator_*` and `0000_70_image-registry-operator_*` are run in parallel.
```

Comment:
They are parallel graph-node entries (it might be worth bringing over [this graph][1] to clarify graph nodes).  That means they could run in parallel, but they might not; it depends on how many workers are available.

Also, let's use some modern examples, like:

```console
$ oc adm release extract --to=manifests quay.io/openshift-release-dev/ocp-release:4.7.2-x86_64
Extracted release payload from digest sha256:83fca12e93240b503f88ec192be5ff0d6dfe750f81e8b5ef71af991337d7c584 created at 2021-03-10T11:32:57Z
$ ls manifests/0000_70_* | cat
manifests/0000_70_cluster-network-operator_00_namespace.yaml
...
manifests/0000_70_cluster-network-operator_05_clusteroperator.yaml
manifests/0000_70_console-operator.crd.yaml
manifests/0000_70_dns-operator_00-cluster-role.yaml
...
manifests/0000_70_dns-operator_03-cluster-operator.yaml
```

So:

>  ..., i.e. `0000_70_cluster-network-operator_*` and `0000_70_dns-operator_*` are parallel graph nodes, and may be run in parallel.

[1]: https://github.com/openshift/cluster-version-operator/blob/69099669668f533fa6659a85c26d607505d8d558/docs/user/reconciliation.md#manifest-graph

Reply:
> They are parallel graph-node entries (it might be worth bringing over [this graph](https://github.com/openshift/cluster-version-operator/blob/69099669668f533fa6659a85c26d607505d8d558/docs/user/reconciliation.md#manifest-graph) to clarify graph nodes). That means they could run in parallel, but they might not; it depends on how many workers are available.

How much of that doc should move over?  Read through, it actually answers question I've had.

Reply:
Also, if the expectation is covering the API around operator-provided manifests, we should include the 0000 defaulting stuff [from `oc`][1]:

> Operators are expected to host the config they need to be installed to a cluster in the `/manifests` directory in their image. This command iterates over a set of operator images and extracts those manifests into a single, ordered list of Kubernetes objects that can then be iteratively updated on a cluster by the cluster version operator when it is time to perform an update. Manifest files are renamed to `0000_70_<image_name>_<filename>` by default, and an operator author that needs to provide a global-ordered file (before or after other operators) should prepend `0000_NN_<component>_` to their filename, which instructs the release builder to not assign a component prefix. Only images in the input that have the image label `io.openshift.release.operator=true` will have manifests loaded.

Because folks who don't care about their order in the payload can reduce some mental overhead by leaving `0000_NN_<component>_` off and relying on `oc` to inject it for them.

[1]: https://github.com/openshift/oc/blob/bdd8f89997ce224cb6f186716c18bee872d3b318/pkg/cli/admin/release/new.go#L73-L82

Reply:
> How much of that doc should move over? Read through, it actually answers question I've had.

That doc is aimed at users trying to understand how the CVO works.  That's distinct from, but not completely orthogonal to the stuff that operator maintainers need to understand to set up their manifests.  Not clear to me whether you want to attempt a single doc aimed at both, or you're scoping this to operator maintainers, or scoping this to just admins, or what.